clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1027,6 +1035,19 @@ static QualType calculateKernelNameType(ASTContext &Ctx,
   return TAL->get(0).getAsType().getCanonicalType();
 }
 
+// Kernel names are currently mangled as type names which
```

Comment:
@AlexeySachkov, I think we might.  Thanks for pointing it out.

Reply:
@premanandrao, thinking about it more, I don't think that it will be the case. Even in 1-pass compiler approach, we will bundle device code as a binary blob into the host application. Here, however, we are embedding another portion of the host code, which contains the same symbols into the app, which causes conflicts.